handbook/unity.adoc

```diff
@@ -0,0 +1,158 @@
+
+An illustration of the general overview of such a receiver system is shown in <<fig_unityoverview>> and few terminologies might help for further discussion:
+
+* *Transmitter*: A node which transmits information
+
+* *Main receiver*: A node that acts as the main receiver in a group of receiver nodes. It is the main receiver’s responsibility to decode the information.
+
+* *Assisting receiver*: A node that acts as an assisting receiver in a group of receiver nodes. It is the assisting receiver’s responsibility to forward the relevant information to the main receiver.
+
+The assisting receiver nodes are in cahoots with the main receiver to cooperatively share the information. This sharing of information usually happens over a short-range wired or wireless network (e.g., WiFi, TCP/IP, UDP/IP). Although, nothing stops one from using a different technology for sharing information.
+
+=== Immediate practical advantage
+
+Consider a case where you are on a ship with a modem deployed and you’ve been receiving status updates from a deployed AUV every few minutes. The AUV moves into an area where the connectivity is poor, and you can no longer successfully receive the status reports. We’ve all experienced being in that situation where the underwater communication link breaks, haven’t we?
+
+If you have a second modem available on the ship, you deploy it from another part of the ship. Or maybe there is a gateway buoy deployed nearby with a modem, and you can connect to its modem. Either way, the distributed spatial diversity technique (Unity) magically uses the information from both modems to recover connectivity to the AUV!
+
+=== How to use Unity in UnetStack3 ?
+
+Unity is available as a premium agent and requires UnetStack 3.1 and higher. Configuring and using `Unity` agent to exploit spatial diversity is easy with just two simple steps:
+
+1. Setup the receiver nodes to cooperate.
+2. Add the Unity agent on the main receiver.
+
+Before we get into the details of how to set it up, let us start a simulated network with 3 nodes by simply running the `netq-network.groovy` simulation script. In the following description, we will setup the Node C to be the main receiver, Node B to be the assisting receiver and Node A to be the transmitter.
+
+==== Set up for receiver nodes to cooperate
+
+To set up the group of receivers to cooperate over a short-range network, we use Wormhole (<<Wormholes>>) agent provided in UnetStack. Transmitter node makes a transmission that is heard at all the receiver nodes. However, it is possible that none of the nodes are able to successfully recover the information received, as the communication link is noisy. In order to share the received noisy signals among the receivers, we can connect the receiver nodes using a Wormhole. A UDP connection between the two receiver nodes, over any IP based network (Ethernet, WiFi), can be established by adding just a few lines of code on the receiver nodes as shown below:
```

Comment:
Provided the receiver nodes are connected using a faster network.

Reply:
Yup. Agreed. Nothing to change for this one since we have already talked about it